src/mad_spark_alt/core/llm_provider.py

```
@@ -609,8 +629,8 @@ def get_available_models(self) -> List[ModelConfig]:
     def _get_default_model_config(self, model_name: str) -> ModelConfig:
         """Get default model config by name."""
         models = {model.model_name: model for model in self.get_available_models()}
-        # Use gemini-1.5-flash as fallback for better stability
-        fallback = models.get("gemini-1.5-flash", list(models.values())[0])
+        # Use gemini-2.5-flash as fallback for latest capabilities
+        fallback = models.get("gemini-2.5-flash", list(models.values())[0])
```

Ruff (0.11.9)

Prefer `next(iter(models.values()))` over single element slice

Replace with `next(iter(models.values()))`

(RUF015)